sklearn/gaussian_process/gpr.py

```diff
@@ -320,13 +320,18 @@ def predict(self, X, return_std=False, return_cov=False):
                 y_cov = self.kernel_(X) - K_trans.dot(v)  # Line 6
                 return y_mean, y_cov
             elif return_std:
-                # compute inverse K_inv of K based on its Cholesky
-                # decomposition L and its inverse L_inv
-                L_inv = solve_triangular(self.L_.T, np.eye(self.L_.shape[0]))
-                K_inv = L_inv.dot(L_inv.T)
+                # cache result of K_inv computation
+                if not hasattr(self, '_K_inv'):
+                    # compute inverse K_inv of K based on its Cholesky
+                    # decomposition L and its inverse L_inv
+                    L_inv = solve_triangular(self.L_.T,
+                                             np.eye(self.L_.shape[0]))
+                    self._K_inv = L_inv.dot(L_inv.T)
```

Comment:
`self.L_` is changed only in [this line](https://github.com/kevinlmh/scikit-learn/blob/cef05eb7fca442c40063a3f0dfe8d15ece6922bd/sklearn/gaussian_process/gpr.py#L247), so you should add for example `self._K_inv = None`just after, and change the condition to `if self._K_inv is None`